### PR TITLE
Muon - Widget; Load Widget

### PR DESCRIPTION
**Pull request by TWJubb**
Refs #23642 

Requires #23670 #23671 and #23672 to be merged.

This PR creates the top-level load widget for the new Muon interface. It combines the load run and load file widgets in the required list above. 

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/Muon/GUI/MuonAnalysis/load_widget/load_widget_presenter.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+    def handle_multiple_files_option_changed(self, unused=None):
+        if self._view.get_multiple_loading_state():
+            self.enable_multiple_files(True)
+        else:
+            self.enable_multiple_files(False)
+
+        selection = self._view.get_multiple_loading_combo_text()
+
+        if selection == "Co-Add":
```
Comment (AnthonyLim23) [inline scripts/Muon/GUI/MuonAnalysis/load_widget/load_widget_presenter.py]: lets make this a constant at the top along with simultaneous

File: scripts/Muon_Analysis_2.py [hunk 2]
```diff
@@ -11,21 +11,82 @@
...
+from Muon.GUI.MuonAnalysis.load_widget.load_widget import LoadWidget
+import Muon.GUI.Common.message_box as message_box
+from Muon.GUI.Common.muon_load_data import MuonLoadData
+
 
 Name = "Muon_Analysis_2"
 
+
 muonGUI = None
+SUPPORTED_FACILITIES = ["ISIS", "SmuS"]
```
Comment (AnthonyLim23) [inline scripts/Muon_Analysis_2.py]: what is SmuS?

File: scripts/Muon_Analysis_2.py [hunk 3]
```diff
@@ -11,21 +11,82 @@
...
+    """
+    current_facility = ConfigServiceImpl.Instance().getFacility().name()
+    if current_facility not in SUPPORTED_FACILITIES:
+        raise AttributeError("Your facility {} is not supported by MuonAnalysis 2.0, so you"
+                             "will not be able to load any files. \n \n"
+                             "Supported facilities are :"
+                             + "\n - ".join(SUPPORTED_FACILITIES))
+
+
+class MuonAnalysis4Gui(QtGui.QMainWindow):
```
Comment (AnthonyLim23) [inline scripts/Muon_Analysis_2.py]: 4?

File: scripts/Muon_Analysis_2.py [hunk 4]
```diff
@@ -11,21 +11,82 @@
...
+        self.load_widget = LoadWidget(self.loaded_data, self.context.instrument, self)
+
+        splitter = QtGui.QSplitter(QtCore.Qt.Vertical)
+        splitter.addWidget(self.load_widget.load_widget_view)
+
+        self.setCentralWidget(splitter)
+        self.setWindowTitle("Muon Analysis version 2")
+
+    def closeEvent(self, event):
+        print("Muon Analysis Close Event")
```
Comment (AnthonyLim23) [inline scripts/Muon_Analysis_2.py]: would `self.load_widget = None` do all of the below too?

File: scripts/Muon_Analysis_2.py [hunk 5]
```diff
@@ -110,9 +170,11 @@ def saveToProject():
 def loadFromProject(project):
     global muonGUI
     muonGUI = main()
-    muonGUI.loadFromContext(project)
+    muonGUI.dock_widget.loadFromProject(project)
+    # muonGUI.loadFromContext(project)
```
Comment (AnthonyLim23) [inline scripts/Muon_Analysis_2.py]: why have you commented this out?

File: scripts/Muon_Analysis_2.py [hunk 6]
```diff
@@ -110,9 +170,11 @@ def saveToProject():
...
     return muonGUI
 
+
```
Comment (AnthonyLim23) [inline scripts/Muon_Analysis_2.py]: does this mean that if I checkout this branch I can then load files into the ADS?

File: scripts/test/Muon/loading_tab/loadwidget_presenter_multiple_file_test.py [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+from Muon.GUI.MuonAnalysis.loadfile.load_file_view import BrowseFileWidgetView
+
+from Muon.GUI.Common.muon_load_data import MuonLoadData
+
+# global QApplication (get errors if > 1 instance in the code)
+QT_APP = QApplication([])
+
+
+class LoadRunWidgetPresenterMultipleFileTest(unittest.TestCase):
+    class Runner:
```
Comment (AnthonyLim23) [inline scripts/test/Muon/loading_tab/loadwidget_presenter_multiple_file_test.py]: did we split the runner out in a different PR?

File: scripts/test/Muon/loading_tab/loadwidget_presenter_multiple_file_test.py [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+
+    def mock_user_input_multiple_files(self, workspaces, filenames, runs):
+        self.load_file_view.get_file_edit_text = mock.Mock(
+            return_value=";".join(filenames))
+        self.load_file_model.load_workspace_from_filename = mock.Mock(
+            side_effect=zip(workspaces, runs))
+
+    def assert_model_contains_correct_loaded_data(self):
+        # use sorted due to threads finishing at different times
+        self.assertEqual(sorted(self.presenter._model.filenames), sorted(self.filenames))
```
Comment (AnthonyLim23) [inline scripts/test/Muon/loading_tab/loadwidget_presenter_multiple_file_test.py]: where did we load some files?

File: scripts/test/Muon/loading_tab/loadwidget_presenter_multiple_file_test.py [hunk 9]
```diff
@@ -0,0 +1,143 @@
...
+        self.assertEqual(sorted(self.presenter._model.workspaces), sorted(self.workspaces))
+        self.assertEqual(sorted(self.presenter._model.runs), sorted(self.runs))
+
+    def assert_interface_contains_correct_runs_and_files(self):
+        self.assertEqual(self.load_file_view.get_file_edit_text(), ";".join(self.filenames))
+        self.assertEqual(self.load_run_view.get_run_edit_text(), "1234-1243")
+
+    def mock_disabling_buttons_in_run_and_file_widget(self):
```
Comment (AnthonyLim23) [inline scripts/test/Muon/loading_tab/loadwidget_presenter_multiple_file_test.py]: minor point, but would this be better above the asserts so all of the mock functions are together?

File: scripts/test/Muon/loading_tab/loadwidget_presenter_test.py [hunk 10]
```diff
@@ -0,0 +1,358 @@
...
+        self.assertEqual(self.load_file_view.get_file_edit_text(), "C:\dir1\dir2\dir3\EMU0001234.nxs")
+        self.assertEqual(self.load_run_view.get_run_edit_text(), "1234")
+
+    def test_that_loading_via_browse_disables_all_load_buttons(self):
+        self.mock_disabling_buttons_in_run_and_file_widget()
+
+        self.presenter.load_file_widget.on_browse_button_clicked()
+        self.Runner(self.presenter.load_file_widget._load_thread)
+
+        self.assertEqual(self.load_file_view.disable_load_buttons.call_count,
```
Comment (AnthonyLim23) [inline scripts/test/Muon/loading_tab/loadwidget_presenter_test.py]: we should also check the value is 1 (if they are both zero the test should fail but will pass)

File: scripts/test/Muon/loading_tab/loadwidget_presenter_test.py [hunk 11]
```diff
@@ -0,0 +1,358 @@
...
+
+    def test_that_loading_via_user_input_file_disables_all_load_buttons(self):
+        self.mock_user_input_single_file([1], "C:\dir1\dir2\dir3\EMU0001111.nxs", 1111)
+
+        self.mock_disabling_buttons_in_run_and_file_widget()
+
+        self.presenter.load_file_widget.handle_file_changed_by_user()
+        self.Runner(self.presenter.load_file_widget._load_thread)
+
+        self.assertEqual(self.load_file_view.disable_load_buttons.call_count,
```
Comment (AnthonyLim23) [inline scripts/test/Muon/loading_tab/loadwidget_presenter_test.py]: same as above

File: scripts/test/Muon/loading_tab/loadwidget_presenter_test.py [hunk 12]
```diff
@@ -0,0 +1,358 @@
...
+        self.presenter.load_run_widget.handle_run_changed_by_user()
+        self.Runner(self.presenter.load_run_widget._load_thread)
+
+        self.assertEqual(self.load_file_view.disable_load_buttons.call_count,
+                         self.load_file_view.enable_load_buttons.call_count)
+        self.assertEqual(self.load_run_view.disable_load_buttons.call_count,
+                         self.load_run_view.enable_load_buttons.call_count)
+
+
+class LoadRunWidgetPresenterLoadFailTest(unittest.TestCase):
```
Comment (AnthonyLim23) [inline scripts/test/Muon/loading_tab/loadwidget_presenter_test.py]: should this be its own file?

**Comment by AnthonyLim23** [comment]
currently defaults to IRIS, could we change this to EMU? Also the names are not very good, will we change the naming conventions etc in a later PR?

File: scripts/Muon_Analysis_2.py (hunk 2 above)
Comment (Mantid-Matthew) [reply scripts/Muon_Analysis_2.py]: I assumed it was some other Muon facility, if you don't know what it is shall I just remove it?

File: scripts/Muon_Analysis_2.py (hunk 3 above)
Comment (Mantid-Matthew) [reply scripts/Muon_Analysis_2.py]: I've tidied this function up by removing the multiple versions

File: scripts/Muon_Analysis_2.py (hunk 4 above)
Comment (Mantid-Matthew) [reply scripts/Muon_Analysis_2.py]: Yes I think so.

File: scripts/Muon_Analysis_2.py (hunk 5 above)
Comment (Mantid-Matthew) [reply scripts/Muon_Analysis_2.py]: I'm uncertain, the new GUI doesn't currently use this code. I have uncommented for now.

File: scripts/Muon_Analysis_2.py (hunk 6 above)
Comment (Mantid-Matthew) [reply scripts/Muon_Analysis_2.py]: I have put in a temporary line to ensure that files are loaded into the ADS. This is purely for testing purposes.

File: scripts/Muon_Analysis_2.py (hunk 2 above)
Comment (AnthonyLim23) [reply scripts/Muon_Analysis_2.py]: see https://muonsources.org/muon-centres.html at the moment we only support ISIS but we can in theory support PSI as we have a data loader for it now

File: scripts/Muon_Analysis_2.py (hunk 3 above)
Comment (AnthonyLim23) [reply scripts/Muon_Analysis_2.py]: I thought meant version 4 so found it confusing as we are building version 2

File: scripts/test/Muon/loading_tab/loadwidget_presenter_multiple_file_test.py (hunk 7 above)
Comment (Mantid-Matthew) [reply scripts/test/Muon/loading_tab/loadwidget_presenter_multiple_file_test.py]: Yes we did I have now updated this as well

File: scripts/test/Muon/loading_tab/loadwidget_presenter_multiple_file_test.py (hunk 9 above)
Comment (Mantid-Matthew) [reply scripts/test/Muon/loading_tab/loadwidget_presenter_multiple_file_test.py]: Have moved it up.

File: scripts/test/Muon/loading_tab/loadwidget_presenter_test.py (hunk 10 above)
Comment (Mantid-Matthew) [reply scripts/test/Muon/loading_tab/loadwidget_presenter_test.py]: have done

File: scripts/test/Muon/loading_tab/loadwidget_presenter_test.py (hunk 11 above)
Comment (Mantid-Matthew) [reply scripts/test/Muon/loading_tab/loadwidget_presenter_test.py]: have done so

**Comment by Mantid-Matthew** [comment]
Check multi-period data.

**Comment by AnthonyLim23** [comment]
If I turn on multiple loading and then use co-add, entering 84446-8 causes the GUI to freeze as it mixes single and multi period data. The current GUI appears to just co-add the first period. 

If I do simultaneous with mixed periods it just gives the first period.

**Review by AnthonyLim23 — DISMISSED** [review]

**Comment by martyngigg** [comment]
We've got some conflicts here now from merging one of the other PRs.

**Review by AnthonyLim23 — APPROVED** [review]
